overlord/snapstate/handlers.go

```diff
@@ -310,20 +323,122 @@ func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName string,
 	return ts, err
 }
 
-func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string, userID int, tm timings.Measurer) error {
+// updates a prerequisite, if it's not providing a content interface that a plug expects it to
+func updatePrereqIfOutdated(t *state.Task, snapName string, contentAttrs []string, userID int, flags Flags) (*state.TaskSet, error) {
+	if len(contentAttrs) == 0 {
+		return nil, nil
+	}
+
+	st := t.State()
+
+	// check if the default provider has all expected content tags
+	if ok, err := hasAllContentAttrs(st, snapName, contentAttrs); err != nil {
+		return nil, err
+	} else if ok {
+		return nil, nil
+	}
+
+	// TODO: this check is likely subset of the conflict checking in Update.
+	// Consider removing it
```

Comment:
we should drop the TODO, I looked at the code again but Update basically checks for conflicts only after having consulted the store, that's a slow process, so this is just an optimisation in a sense, but I think we do want it

Reply:
Make sense, I can replace the TODO with a small comment indicating it's an optimization (just to avoid confusion in the future)